### PR TITLE
UPDATE_KOTLIN_VERSION: 1.7.0-dev-3025

### PR DESCRIPTION
**Pull request by ting-yuan**

### CODE REVIEW

File: gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt [hunk 1]
```diff
@@ -244,7 +243,6 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
             )
             kspTask.commandLineArgumentProviders.addAll(kspExtension.commandLineArgumentProviders)
             kspTask.destination = kspOutputDir
-            kspTask.kspOutputDir.value(kspOutputDir)
```
Comment (neetopia) [inline gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt:247]: Is this field deprecated?

File: gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt [hunk 2]
```diff
@@ -277,19 +275,19 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
...
-                kotlinCompilation.allKotlinSourceSets.forEach { sourceSet -> kspTask.source(sourceSet.kotlin) }
+                kotlinCompilation.allKotlinSourceSets.forEach { sourceSet ->
+                    kspTask.setSource(sourceSet.kotlin)
+                }
                 if (kotlinCompilation is KotlinCommonCompilation) {
-                    kspTask.source(kotlinCompilation.defaultSourceSet.kotlin)
+                    kspTask.setSource(kotlinCompilation.defaultSourceSet.kotlin)
                 }
             }
+            kspTask.exclude { kspOutputDir.isParentOf(it.file) }
```
Comment (neetopia) [inline gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt:290]: What's the use of this?

**Review by neetopia — APPROVED** [review]
LGTM, with 2 questions regarding the update.

File: gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt (hunk 1 above)
Comment (ting-yuan) [reply gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt:247]: It's actually a duplication to the above line.

File: gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt (hunk 2 above)
Comment (ting-yuan) [reply gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt:290]: Good catch. It's a leftover from rebase and is removed.

**Pull request merged**